examples/PyTorch/Inference/hf_transformers/blade_adapter.py

```diff
@@ -0,0 +1,305 @@
+
+
+def _kwargs_to_args(input_order, **kwargs) -> Tuple:
+    return tuple(kwargs[k] for k in input_order)
+
+
+def _default_device() -> torch.device:
+    return torch.device('cuda:0') if torch.cuda.is_available() else torch.device('cpu')
+
+
+class BladeModel(PreTrainedModel):
+    def __init__(self, opt_model: torch.jit.ScriptModule, info: ModelInfo) -> None:
+        super().__init__(info.config)
+
+        self.opt_model = opt_model
+        self.info = info
+
+    def forward(self, *args, **kwargs) -> Any:
+        # TODO(litan.ls): merge kwargs to args according to tracable wrapper
+        model_args = list(args)
+        model_args.extend(_kwargs_to_args(self.info.input_order, **kwargs))
+        outputs = self.opt_model(*model_args)
+        return ModelOutput({k: v for k, v in zip(self.info.output_names, outputs)})
+
+
+def create_model(task: Optional[str] = None, id_or_path: Optional[str] = None,
+                 preprocessor: Optional[Preprocessor] = None,
+                 **model_kwargs) -> Tuple[PreTrainedModel, ModelInfo]:
+    info = load_model_info(task, id_or_path, preprocessor=preprocessor)
+    _, model = infer_framework_load_model(info.id_or_path, info.config, model_classes=info.targeted_task,
+                                          task=info.task, framework='pt', **model_kwargs)
+    return model, info
+
+
+def optimize(task: Optional[str] = None, id_or_path: Optional[str] = None,
+             model: Optional[PreTrainedModel] = None,
+             preprocessor: Optional[Preprocessor] = None,
+             skip_compile: bool = False, amp: bool = False,
+             device: Optional[torch.device] = None,
+             model_kwargs: Dict[str, Any] = None) -> BladeModel:
+    if device is None:
+        device = _default_device()
+
+    if model is None:
+        model_kwargs = model_kwargs or {}
+        model_kwargs['torchscript'] = True
+        model, info = create_model(task=task, id_or_path=id_or_path,
+                                   preprocessor=preprocessor, **model_kwargs)
+    else:
+        if task is None:
+            raise ValueError('Must specify task when optimize loaded model.')
+        if not model.config.torchscript:
+            raise ValueError(
+                'Only support optimize model with torchscript=True.')
+        info = load_model_info(
+            task=task, config=model.config, preprocessor=preprocessor)
+
+    model.to(device)
+    for k, v in info.dummy_inputs.items():
+        info.dummy_inputs[k] = v.to(device)
+
+    tracable = Tracable(model, [], info.default_kwargs, amp=amp)
+    traced = torch.jit.trace(tracable.eval(), _kwargs_to_args(
+        info.input_order, **info.dummy_inputs), strict=False)
+
+    if skip_compile:
+        opt_model = traced
+    else:
+        # TODO(litan.ls): support custom blade config
+        torch_config = torch_blade.config.Config()
+        torch_config.enable_mlir_amp = False  # disable mix-precision
```

Comment:
The default value of `enable_mlir_amp` is False, there is no need to set it explicitly

Reply:
Removed it.